lib/slack/web/faraday/connection.rb

```diff
@@ -22,11 +22,14 @@ def connection
               options[:request] = request_options if request_options.any?
 
               ::Faraday::Connection.new(endpoint, options) do |connection|
-                connection.use ::Faraday::Request::Multipart
-                connection.use ::Faraday::Request::UrlEncoded
+                connection.request :multipart
+                connection.request :url_encoded
                 connection.use ::Slack::Web::Faraday::Response::RaiseError
-                connection.use ::FaradayMiddleware::Mashify, mash_class: Slack::Messages::Message
-                connection.use ::FaradayMiddleware::ParseJson
+                connection.response :mashify, mash_class: Slack::Messages::Message
+                # Match any content type for JSON responses meaning Faraday::ParsingError will be raised for non-JSON
+                # responses such as HTML 500 errors.
+                # TODO: Should this be more restrictive?
+                connection.response :json, content_type: /\b*$/
```

Comment:
So this is any content type, isn't it? Do any specs fail if you remove `content_type` here? I think not, and it's the old behavior that basically says "try to parse anything, which now raises ::Faraday::ParsingError, which we handle".

Reply:
If we remove this `content_type:` option then the `<html></html>` response specs will fail because the default for the [JSON middleware](https://lostisland.github.io/faraday/middleware/json-response) parsing is `/\bjson$/`, so we wouldn't attempt to parse if a 500 HTML response was returned and we end up with the `undefined method 'map' for nil:NilClass` mention in the initial PR description.

I guess there are two options:

1) Leave as is it now meaning that it will try and parse any response Slack returns, which I think as you say is the old behaviour handled by the deprecated `::FaradayMiddleware::ParseJson` middleware
2) Make the content types stricter and set it to only attempt to parse JSON or HTML content types as we can set an array of content types

Reply:
> Remove comment and TODO above.

Removed 👍🏻